internal/controller/api/http_proxy_connection_manager.go

```diff
@@ -52,82 +50,45 @@ func (rcl *RedisConnectionLocator) GetConnection(account string, node_id string)
 
 func (rcl *RedisConnectionLocator) GetConnectionsByAccount(account string) map[string]controller.Receptor {
 
-	var cursor uint64
-	var err error
-
 	connectionsPerAccount := make(map[string]controller.Receptor)
 
-	connections := make(map[string]interface{})
-
-	for {
-		var keys []string
-		pattern := fmt.Sprintf("%s:*", account)
-		if keys, cursor, err = rcl.RedisConnection.Scan(cursor, pattern, 50).Result(); err != nil {
-			// FIXME: log error, return an error
-			return connectionsPerAccount
-		}
-
-		if len(keys) > 0 {
-			fmt.Printf("found %d keys\n", len(keys))
-		}
-
-		for _, key := range keys {
-			connections[key] = nil
-		}
-
-		if cursor == 0 {
-			break
-		}
+	accountConnections, err := controller.GetRedisConnectionsByAccount(rcl.Client, account)
+	if err != nil {
+		// FIXME: Update connectionlocator interface methods to return error
+		logger.Log.Warnf("Error during lookup for account: %s", account)
+		return nil
 	}
 
-	for key, _ := range connections {
-		s := strings.Split(key, ":")
-		proxy := rcl.GetConnection(s[0], s[1])
-		connectionsPerAccount[key] = proxy
+	for _, conn := range accountConnections {
+		s := strings.Split(conn, ":")
+		nodeID := s[0]
+		proxy := rcl.GetConnection(account, nodeID)
+		// FIXME: Is this a good key?
+		connectionsPerAccount[account+":"+nodeID] = proxy
 	}
 
 	return connectionsPerAccount
 }
 
 func (rcl *RedisConnectionLocator) GetAllConnections() map[string]map[string]controller.Receptor {
 
-	var cursor uint64
-	var err error
-
 	connectionMap := make(map[string]map[string]controller.Receptor)
 
-	connections := make(map[string]interface{})
-
-	for {
-		var keys []string
-		if keys, cursor, err = rcl.RedisConnection.Scan(cursor, "*", 50).Result(); err != nil {
-			// FIXME: log error, return an error
-			return connectionMap
-		}
-
-		if len(keys) > 0 {
-			fmt.Printf("found %d keys\n", len(keys))
-		}
-
-		for _, key := range keys {
-			connections[key] = nil
-		}
-
-		if cursor == 0 {
-			break
-		}
+	connections, err := controller.GetAllRedisConnections(rcl.Client)
+	if err != nil {
+		// FIXME: Update connectionlocator interface methods to return error
+		logger.Log.Warn("Error during lookup for all connections")
+		return nil
 	}
 
-	for key, _ := range connections {
-		s := strings.Split(key, ":")
-		account := s[0]
-		nodeId := s[1]
-		proxy := rcl.GetConnection(account, nodeId)
-		_, exists := connectionMap[account]
-		if exists == false {
+	for _, conn := range connections {
+		s := strings.Split(conn, ":")
```

Comment:
Like the other method, I think we are leaking details here.

Reply:
:+1: